src/test/java/uk/gov/companieshouse/api/accounts/service/impl/CompanyAccountServiceImplTest.java

```diff
@@ -2,54 +2,116 @@
 import uk.gov.companieshouse.api.accounts.model.entity.CompanyAccountEntity;
 import uk.gov.companieshouse.api.accounts.model.rest.CompanyAccount;
 import uk.gov.companieshouse.api.accounts.repository.CompanyAccountRepository;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionManager;
+import uk.gov.companieshouse.api.accounts.transaction.TransactionStatus;
 import uk.gov.companieshouse.api.accounts.transformer.CompanyAccountTransformer;
 
 @ExtendWith(MockitoExtension.class)
 @TestInstance(Lifecycle.PER_CLASS)
 public class CompanyAccountServiceImplTest {
 
+    @InjectMocks
+    private CompanyAccountServiceImpl companyAccountService;
+
+    @Mock
+    private CompanyAccount companyAccountMock;
+
     @Mock
-    private CompanyAccount companyAccount;
+    private CompanyAccountEntity companyAccountEntityMock;
 
     @Mock
-    private CompanyAccountEntity createdCompanyAccountEntity;
+    private CompanyAccountDataEntity companyAccountDataEntityMock;
 
     @Mock
     private CompanyAccountRepository companyAccountRepository;
 
     @Mock
     private CompanyAccountTransformer companyAccountTransformer;
 
-    @InjectMocks
-    private CompanyAccountServiceImpl companyAccountService;
+    @Mock
+    private TransactionManager transactionManagerMock;
+
+    @Mock
+    private HttpServletRequest httpServletRequest;
+
+    @Mock
+    private HttpSession httpSessionMock;
 
     @BeforeEach
-    public void setUp() {
-        when(companyAccountTransformer.transform(companyAccount)).thenReturn(createdCompanyAccountEntity);
+    void setUp() {
+        RequestContextHolder.setRequestAttributes(new ServletRequestAttributes(httpServletRequest));
+
+        when(httpServletRequest.getSession()).thenReturn(httpSessionMock);
+        when(httpSessionMock.getAttribute("transaction")).thenReturn(createDummyTransaction(true));
+        when(httpServletRequest.getHeader("X-Request-Id")).thenReturn("test");
+        when(companyAccountDataEntityMock.getLinks()).thenReturn(createLinksMap());
     }
 
     @Test
-    @DisplayName("Tests the successful creation of an companyAccount resource")
-    public void canCreateAccount() throws NoSuchAlgorithmException {
-        CompanyAccount result = companyAccountService.save(companyAccount, "");
+    @DisplayName("Tests the successful creation of an company account resource")
+    void canCreateAccount() {
+        doReturn(companyAccountEntityMock).when(companyAccountTransformer).transform(ArgumentMatchers
+                .any(CompanyAccount.class));
+
+        when(companyAccountEntityMock.getData()).thenReturn(companyAccountDataEntityMock);
+
+        CompanyAccount result = companyAccountService.createCompanyAccount(companyAccountMock);
+  
         assertNotNull(result);
-        assertEquals(companyAccount, result);
+        assertEquals(companyAccountMock, result);
```

Comment:
This test is titled that it tests the successful creation of the resource but it doesn't verify that the resource was created.

e.g.
```java
// set up mocks
// run test

assertNotNull(result);
assertEquals(companyAccountMock, result);
verify(companyAccountRepository.save(companyAccountEntityMock));
```

Reply:
done